overlord/snapstate/target.go

```diff
@@ -63,30 +63,30 @@ type Options struct {
 	RequireOneSnap bool
 }
 
-// Target represents the data needed to setup a snap for installation.
-type Target struct {
-	// Snap is a partially initialized SnapSetup that contains the data needed
+// target represents the data needed to setup a snap for installation.
+type target struct {
+	// setup is a partially initialized SnapSetup that contains the data needed
 	// to find the snap file that will be installed.
-	Snap *SnapSetup
-	// Components is a list of partially initialized ComponentSetup structs that
+	setup *SnapSetup
+	// components is a list of partially initialized ComponentSetup structs that
 	// contain the data needed to find the component files that will be
 	// installed.
-	Components []*ComponentSetup
-	// Info contains the snap.Info for the snap to be installed.
-	Info *snap.Info
-	// SnapState is the current state of the target snap, prior to installation.
+	components []*ComponentSetup
+	// info contains the snap.info for the snap to be installed.
+	info *snap.Info
+	// snapst is the current state of the target snap, prior to installation.
 	// This must be retrieved prior to unlocking the state for any reason (for
 	// example, talking to the store).
-	SnapState SnapState
+	snapst SnapState
 }
 
-// InstallGoal represents a single snap or a group of snaps to be installed.
-type InstallGoal interface {
-	// ToInstall returns the data needed to setup the snaps for installation.
-	ToInstall(context.Context, *state.State, Options) ([]Target, error)
+// installGoal represents a single snap or a group of snaps to be installed.
+type installGoal interface {
```

Comment:
this one should probably stay exported, is usually strange to have input to an exported  function that are unexported